br/pkg/lightning/restore/table_restore.go

```diff
@@ -712,44 +712,47 @@ func (tr *TableRestore) postProcess(
 				localChecksum.Add(&chunk.Checksum)
 			}
 		}
+		tr.logger.Info("local checksum", zap.Object("checksum", &localChecksum))
+
+		// 4.5. do duplicate detection.
+		hasDupe := false
+		if rc.cfg.TikvImporter.DuplicateDetection {
```

Comment:
Does this check will significantly affect the performance, since in most cast there should be no duplicate

Reply:
Local duplicate collection is basically free if there's no dupes.

Remote duplicate collection requires an O(n) scan of any imported data.

Reply:
So I prefer to only do (remote) duplicate check only if checksum failed and (maybe) only if remoteKvs < localKvs.

Reply:
checksum passing is not a certain way of saying there's no duplicates though. checksum has no false negatives but does have false positives.